app/src/main/java/com/android/voyageur/ui/trip/activities/ActivityUtils.kt

```diff
@@ -63,3 +74,52 @@ fun DeleteActivityAlertDialog(
       },
       dismissButton = { TextButton(onClick = { onDismissRequest() }) { Text("Cancel") } })
 }
+/**
+ * A dialog for filtering activities by type. Displays a list of available activity types with
+ * checkboxes to toggle filters.
+ *
+ * @param selectedFilters A set of currently selected [ActivityType] filters.
+ * @param onFilterChanged Callback invoked when a filter checkbox is checked or unchecked.
+ * @param onDismiss Callback invoked to close the dialog.
+ */
+@Composable
+fun FilterDialog(
+    selectedFilters: Set<ActivityType>,
+    onFilterChanged: (ActivityType, Boolean) -> Unit,
+    onDismiss: () -> Unit
+) {
+  AlertDialog(
+      onDismissRequest = { onDismiss() },
+      modifier = Modifier.testTag("filterActivityAlertDialog"),
+      title = {
+        Text(
+            text = stringResource(R.string.filter_activity_type),
+            style = MaterialTheme.typography.titleMedium)
+      },
+      text = {
+        Column(
+            modifier = Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(10.dp)) {
+              ActivityType.entries.forEach { type ->
+                // A row represents the activity type and a checkbox to validate the filter
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    verticalAlignment = Alignment.CenterVertically,
+                    horizontalArrangement = Arrangement.SpaceBetween) {
+                      Text(
+                          text = type.name,
```

Comment:
Wdyt about changing this to capital letter and then small letters instead of all capital letters? (Restaurant instead of RESTAURANT)
I believe it would look a bit cleaner, but I leave this up to you :)

Reply:
I think you're right as well here! Feels a bit friendlier to have lowercase instead of capital letters. I'll modify this as well.